app/models/event.rb

```diff
@@ -113,20 +121,32 @@ def average_rating
     @total_rating > 0 ? number_with_precision(@total_rating / @total.to_f, precision: 2, strip_insignificant_zeros: true) : 0
   end
 
-  def submitter
-    result = event_users.where(event_role: 'submitter').first
-    if result.nil?
-      user = nil
-      # Perhaps the event_users haven't been saved, if this is a new proposal
-      event_users.each do |u|
-        if u.event_role == 'submitter'
-          user = u.user
-        end
-      end
-      user
-    else
-      result.user
+  # def submitter
+  #   result = event_users.where(event_role: 'submitter').first
+  #   if result.nil?
+  #     user = nil
+  #     # Perhaps the event_users haven't been saved, if this is a new proposal
+  #     event_users.each do |u|
+  #       if u.event_role == 'submitter'
+  #         user = u.user
+  #       end
+  #     end
+  #     user
+  #   else
+  #     result.user
+  #   end
+  # end
+
+  # get event speakers with the event sumbmitter at the first position
+  # if the submitter is also a speaker for this event
+  def speakers_ordered
+    @speakers = speakers.to_a
+    @speakers << submitter unless @speakers.any?
+
+    if @speakers.reject! { |speaker| speaker == submitter }
+      @speakers.insert(0, submitter)
```

Comment:
> the one who submitted an event and who also is a speaker should be shown first

On the other hand, why should one speaker be "more important" than another? :)
Personally, I would keep things simple and order speakers alphabetically, but I won't insist.